src/Microsoft.Sbom.Api/Output/Telemetry/TelemetryRecorder.cs

```diff
@@ -46,6 +47,66 @@ public TelemetryRecorder(IFileSystemUtils fileSystemUtils, IConfiguration config
             Log = log ?? throw new ArgumentNullException(nameof(log));
         }
 
+        private TelemetryRecorder(string telemetryFilePath, IFileSystemUtils fileSystemUtils)
+        {
+            this.telemetryFilePath = telemetryFilePath;
+            FileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        }
+
+        public static TelemetryRecorder Create(string telemetryFilePath, IFileSystemUtils fileSystemUtils)
+        {
+            return new TelemetryRecorder(telemetryFilePath, fileSystemUtils);
+        }
+
+        //Method to log telemetry in conditions when the tool is not able to start execution of workflow.
+        public async Task ConsoleLogger(Exception e)
+        {
+            var logger = new LoggerConfiguration().WriteTo.Console().CreateLogger();
+            try
+            {
+                //convert exception to IDictionary<string,string> and handle unicode char.
+                var exceptions = new Dictionary<string, string>
+                {
+                    { e.GetType().ToString().Replace("\u0027", "'"), e.Message }
+                };
+
+                // Create the telemetry object.
+                var telemetry = new SBOMTelemetry
+                {
+                    Result = Result.Failure,
+                    Timings = timingRecorders.Select(t => t.ToTiming()).ToList(),
+                    Switches = this.switches,
+                    Exceptions = exceptions
```

Comment:
This is was the main issue causing the circular dependency when trying to solve this problem since in cases like this the program is failing before we have a configuration object. I will look into it again and see if I can somehow pass the arguments being used to the logger without the need for the configuration object.